### ISSUE TITLE
Rounding error with atan(acos(asin(ans)))

### ISSUE THREAD

**Issue by wiiikiii** [numworks/epsilon#6]
https://www.numworks.com/eu/simulator/

tan(cos(sin(9)))
atan(acos(asin(ANS)))

the result is 8.989918

is this result only in simlulation or on hardware also?

**Comment by rhaamo** [comment]
I tried the web sim and with the built one and I can only get "undef"

How to reproduce that ?

![selection_038](https://user-images.githubusercontent.com/8625/29827630-bf9a8214-8cda-11e7-8480-4cc9947b6dd2.png)

**Comment by OLaurent-Maths** [comment]
Wrong order of the functions in the issue description:
asin(acos(atan(ans)))

**Comment by rhaamo** [comment]
Thanks! I'm getting the same with the built simulator.
I will try tomorrow with the real calc if there is the same issue.

**Comment by rhaamo** [comment]
I've done more "tests", sim vs "another calc"
```
sin(9), 0.156434 vs  0.412118
cos(9), 0.987688 vs -0.911130
tan(9), 0.158384 vs -0.452315
```

Using KCalc on KDE, I got the same results as the simulator (web or "desktop app"), maybe this could be related to endianess of the platform.

**Comment by OLaurent-Maths** [comment]
Works fine for me on another calculator (TI)
My guess is that when calling ANS, it tried to parse again the previous result : 0.017455 instead of getting the complex number behind it.
If you just try to do: asin(acos(atan(tan(cos(sin(9))), it works perfectly

**Comment by rhaamo** [comment]
Still imprecise.
A casio responds with "0.4247779608", the sim "9", a single cos(), tan() or sin() is even wrong in the sim.
About the wront re-usage of "Ans" I think this is a separate issue.

**Comment by OLaurent-Maths** [comment]
"A casio responds with "0.4247779608", the sim "9",-> this is normal : your CASIO calculator is set to radian so 0.4247779608 is also correct.

**Comment by Ecco** [comment]
Actual results from a device running `679CEA7`.

`asin(acos(atan(tan(cos(sin(9)))))` gives `9`

and

```
tan(cos(sin(9)))
-> 0.017455
asin(acos(atan(ans)))
-> 8.989918
```

This is indeed a rounding bug.

**Comment by OLaurent-Maths** [comment]
Indeed, when evaluating the 'ans' value (Evaluation<double> * LocalContext::ansValue()) it takes the last result of the calculation store with the precision of 7 digits.

**Comment by madhatter0** [comment]
You can generate this even more simply: sqrt(2) followed by (Ans)^2 is 2.000001.

**Comment by madhatter0** [comment]
This is now the oldest open bug.  I hope OLaurent-Maths has put his finger on the problem, above: that Ans is evaluated with truncated precision.  Please, is there any good reason not to carry forward all the available precision when operating on Ans?  Or is something else going wrong here?

**Comment by artaxxx** [comment]
This rounding bug is fixed in the new version 1.2.0
